hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/push/DelayedEventBusPushStrategy.java

```diff
@@ -228,25 +227,23 @@ private void doDispatch(final List<Event> events, final WrappedSession wrappedSe
             }
         }
 
-        private void fowardBulkEvents(final List<Event> events, final SecurityContext userContext) {
-            final Set<Class<?>> filterBulkEvenTypes = eventProvider.getFilteredBulkEventsType(events);
-
-            for (final Class<?> bulkType : filterBulkEvenTypes) {
-                final List<Event> listBulkEvents = events.stream()
-                        .filter(event -> DelayedEventBusPushStrategy.eventSecurityCheck(userContext, event)
-                                && bulkType.isInstance(event))
-                        .collect(Collectors.toList());
-                if (!listBulkEvents.isEmpty()) {
-                    eventBus.publish(vaadinUI, listBulkEvents);
-                }
-            }
-        }
-
-        private void fowardSingleEvents(final List<Event> events, final SecurityContext userContext) {
-            events.stream()
-                    .filter(event -> DelayedEventBusPushStrategy.eventSecurityCheck(userContext, event)
-                            && eventProvider.getSingleEvents().contains(event.getClass()))
-                    .forEach(event -> eventBus.publish(vaadinUI, event));
+        @SuppressWarnings("unchecked")
+        private List<EventContainer<Event>> groupEvents(final List<Event> events, final SecurityContext userContext,
+                final UIEventProvider eventProvider) {
+
+            return events.stream().filter(event -> DelayedEventBusPushStrategy.eventSecurityCheck(userContext, event))
+                    .collect(Collectors.groupingBy(Event::getClass)).entrySet().stream().map(entry -> {
+                        EventContainer<Event> holder = null;
+                        try {
+                            holder = (EventContainer<Event>) eventProvider.getEvents().get(entry.getKey())
+                                    .getConstructor(List.class).newInstance(entry.getValue());
+                        } catch (final NoSuchMethodException | InstantiationException | IllegalAccessException
```

Comment:
If you don't care about the detailed reason why your reflection call is actually failing, better catch `ReflectiveOperationException` only. This improves readability a lot.